src/node_url.cc

```diff
@@ -75,6 +75,73 @@ void BindingData::Deserialize(v8::Local<v8::Context> context,
   CHECK_NOT_NULL(binding);
 }
 
+std::string EncodePathChars(const std::string& input_str, bool windows) {
```

Comment:
We shouldn't run this function if the input doesn't contain any of the forbidden values. Not running it helps us reduce the extra string allocation.

Reply:
> Not running it helps us reduce the extra string allocation.

I'm not sure that's true, we still need to prepend `file://` at the start